### ISSUE TITLE
Loop vectorization in GraalVM CE JIT compilations

### ISSUE THREAD

**Issue by gergo-** [oracle/graal#5468]
#### TL;DR
Make automatic loop vectorization available for just-in-time (JIT) compilations in GraalVM Community Edition.

#### Goals
Add compiler phases and supporting infrastructure to enable automatic loop vectorization in GraalVM CE JIT compilations. Users will be able to write natural Java loops and obtain efficient SIMD code for supported loop shapes. This will include many loops that are not vectorized by HotSpot's C2 compiler.
See the blog post on [Enhanced Automated Vectorization](https://medium.com/graalvm/enhanced-automated-vectorization-in-graalvm-76cd99925b6d) for some examples of automatic vectorization in GraalVM.

#### Non-Goal
- Enable loop vectorization in ahead-of-time (AOT) compilations.

**Comment by JohnTortugo** [comment]
@gergo- - what are the plans for this issue?

**Comment by gergo-** [comment]
There is no concrete timeline at the moment. https://github.com/oracle/graal/issues/10285 will certainly have to come first, since there is a lot of common infrastructure between autovectorization and Vector API lowering.

**Comment by gergo-** [comment]
Implemented in https://github.com/oracle/graal/pull/14105.
